packages/tasks/test/swap/UniswapV3Swapper.test.ts

```diff
@@ -0,0 +1,506 @@
+                    })
+
+                    context('when the slippage is below the limit', () => {
+                      const hopTokens = []
+                      const hopFees = []
+                      const slippage = fp(0.01)
+                      const expectedAmountOut = amountIn.mul(tokenRate)
+                      const minAmountOut = expectedAmountOut.mul(fp(1).sub(slippage)).div(fp(1))
+                      const fee = 1
+
+                      beforeEach('set max slippage', async () => {
+                        const setDefaultMaxSlippageRole = task.interface.getSighash('setDefaultMaxSlippage')
+                        await authorizer
+                          .connect(owner)
+                          .authorize(owner.address, task.address, setDefaultMaxSlippageRole, [])
+                        await task.connect(owner).setDefaultMaxSlippage(slippage)
+                      })
+
+                      it('executes the expected connector', async () => {
+                        const tx = await task.call(tokenIn.address, amountIn, slippage, fee, hopTokens, hopFees)
+
+                        const connectorData = connector.interface.encodeFunctionData('execute', [
+                          tokenIn.address,
+                          tokenOut.address,
+                          amountIn,
+                          minAmountOut,
+                          fee,
+                          hopTokens,
+                          hopFees,
+                        ])
+
+                        await assertIndirectEvent(tx, smartVault.interface, 'Executed', {
+                          connector,
+                          data: connectorData,
+                        })
+
+                        await assertIndirectEvent(tx, connector.interface, 'LogExecute', {
+                          tokenIn,
+                          tokenOut,
+                          amountIn,
+                          minAmountOut,
+                          fee,
+                          hopTokens,
+                          hopFees,
+                        })
+                      })
+
+                      it('emits an Executed event', async () => {
+                        const tx = await task.call(tokenIn.address, amountIn, slippage, fee, hopTokens, hopFees)
+
+                        await assertIndirectEvent(tx, task.interface, 'Executed')
+                      })
+                    })
+
+                    context('when the slippage is above the limit', () => {
+                      const slippage = fp(0.01)
+                      const hopTokens = []
+                      const hopFees = []
+                      const fee = 1
+
+                      it('reverts', async () => {
+                        await expect(
+                          task.call(tokenIn.address, amountIn, slippage, fee, hopTokens, hopFees)
+                        ).to.be.revertedWith('TaskSlippageAboveMax')
```

Comment:
I usually try to define only the variables that are relevant for the context you're describing, in this case the slippage
```suggestion
                      const slippage = fp(0.01)

                      it('reverts', async () => {
                        await expect(
                          task.call(tokenIn.address, amountIn, slippage, 0, [], [])
                        ).to.be.revertedWith('TaskSlippageAboveMax')
```